core/src/protocols_handler/mod.rs

```diff
@@ -150,9 +150,10 @@ pub trait ProtocolsHandler {
     ///
     /// This method is called by the `Swarm` after each invocation of
     /// [`ProtocolsHandler::poll`] to determine if the connection and the associated
-    /// `ProtocolsHandler`s should be kept alive and if so, for how long.
+    /// `ProtocolsHandler`s should be kept alive as far as this handler is concernced
+    /// and if so, for how long.
     ///
-    /// Returning [`KeepAlive::Now`] indicates that the connection should be
+    /// Returning [`KeepAlive::Now`] indicates that the connection can be
```

Comment:
"can vs should" a bit debatable. The value returned here is authoritative from the point of view of the swarm/nodehandler, so from this aspect it should be "should" or even "will".
However I see your point: since handlers are merged together with other handlers by the user, each individual handler doesn't know if returning `Now` is actually going to close the connection.